### PR TITLE
Feature/re introduce ngc compilation for common ngx

### PR DESCRIPTION
**Pull request by esotrope**
1. Re-added ngc compilation for common-ngx
2. Fixed admin/webapp tests by inlining common lib html templates in the component `ts` files
2. Deleted common-ngx/demo dir as we were not using it.
3. Deleted common-ngx/tsconfig-aot.json, webpack.umd.js and other build comps that aren't needed anymore

### CODE REVIEW

File: admin-webapp/src/main/webapp/package.json [hunk 1]
```diff
@@ -45,7 +45,7 @@
   "devDependencies": {
     "@angular/cli": "^1.2.7",
     "@angular/compiler-cli": "4.1.1",
-    "@sbac/rdw-reporting-common-ngx": "file:../../../../common-ngx",
+    "@sbac/rdw-reporting-common-ngx": "file:../../../../common-ngx/dist/sbac-rdw-reporting-common-ngx-0.0.0.tgz",
```
Comment (bgsavage) [inline admin-webapp/src/main/webapp/package.json]: so I'm guessing npm will then explode this intot he @sbac/rdw-reporting-common-ngx directory?

File: common-ngx/demo/package.json [hunk 2]
```diff
@@ -1,46 +0,0 @@
-{
-  "name": "@sbac/rdw-reporting-common-ngx-demo",
```
Comment (bgsavage) [inline common-ngx/demo/package.json]: Guessing this was just a prototype?

File: common-ngx/package.json [hunk 3]
```diff
@@ -20,29 +20,9 @@
...
-    "prebuild": "rimraf dist tmp",
-    "prebuild:watch": "rimraf dist tmp",
-    "prepublishOnly": "npm run ci",
-    "preversion": "npm run ci",
-    "test": "karma start",
-    "test:watch": "karma start --auto-watch --no-single-run",
-    "tslint": "tslint"
+    "build": "rimraf dist && ngc -p ./tsconfig-ngc.json && copyfiles package.json dist && cd dist && npm pack",
+    "test": "karma start karma.conf.ts",
+    "lint": "tslint 'src/**/*.ts'"
```
Comment (bgsavage) [inline common-ngx/package.json]: 👍

File: common-ngx/src/security/session-expired.component.ts [hunk 4]
```diff
@@ -3,7 +3,17 @@ import { AuthenticationService } from "./authentication.service";
...
+  template: `
+    <div>
+      <h3 class="blue mb-md">{{'common-ngx.session-expired.title' | translate}}</h3>
+      <div class="well">
+        <p>{{'common-ngx.session-expired.message' | translate}}</p>
+        <button class="btn btn-primary" (click)="onOk()">
+          <span>{{'common-ngx.session-expired.ok' | translate}}</span>
+        </button>
+      </div>
+    </div>
```
Comment (bgsavage) [inline common-ngx/src/security/session-expired.component.ts]: We should probably have a rule of when to use template vs. templateUrl.

For this, I'd prefer to keep it as a templateUrl as it's more than a few lines and provides a nice separation of concern.

File: common-ngx/tsconfig-ngc.json [hunk 5]
```diff
@@ -0,0 +1,26 @@
+{
+  "compilerOptions": {
+    "target": "es5",
+    "module": "es2015",
```
Comment (bgsavage) [inline common-ngx/tsconfig-ngc.json]: not es6?

File: common-ngx/tsconfig-ngc.json [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    "stripInternal": true
+  },
+  "exclude": [
+    "node_modules"
+  ],
+  "files": [
+    "./src/index.ts"
+  ],
+  "angularCompilerOptions": {
+    "skipTemplateCodegen": true
```
Comment (bgsavage) [inline common-ngx/tsconfig-ngc.json]: what's the impact of this?

File: common-ngx/demo/package.json (hunk 2 above)
Comment (esotrope) [reply common-ngx/demo/package.json]: yup

File: admin-webapp/src/main/webapp/package.json (hunk 1 above)
Comment (esotrope) [reply admin-webapp/src/main/webapp/package.json]: correctamundo

File: common-ngx/src/security/session-expired.component.ts (hunk 4 above)
Comment (esotrope) [reply common-ngx/src/security/session-expired.component.ts]: ngc requires templates & stylesheets be inlined. originally the gulp.js was taking care of this but for simplicity i took it out and figured we could do that ourselves. so for common lib let's stick to inline templates.
thoughts? concerns?

File: common-ngx/tsconfig-ngc.json (hunk 5 above)
Comment (esotrope) [reply common-ngx/tsconfig-ngc.json]: es6 === es2015
`target: es5` means we compile our es6/ts down to es5 (what all browsers have implemented atm)
`libs: [es6, dom]` the libs we need to compile our es6

File: common-ngx/src/security/session-expired.component.ts (hunk 4 above)
Comment (bgsavage) [reply common-ngx/src/security/session-expired.component.ts]: Hm well that's an interesting constraint.  Isn't that how React works?  I think that would not be ideal if we start having larger templates here.  Also creates an inconsistency in here vs. the dependent apps that don't have that constraint.

File: common-ngx/tsconfig-ngc.json (hunk 6 above)
Comment (esotrope) [reply common-ngx/tsconfig-ngc.json]: > By default ngc generates ngfactories for the components and modules. By using skipTemplateCodegen flag we can skip this and only get *.metadata.json files.
http://blog.mgechev.com/2017/01/21/distributing-an-angular-library-aot-ngc-types/

**Pull request merged**